data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v3/GetNewBlockV3IntegrationTest.java

```diff
@@ -0,0 +1,328 @@
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.header(HEADER_CONSENSUS_VERSION))
+        .isEqualTo(specMilestone.name().toLowerCase(Locale.ROOT));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_BLINDED)).isEqualTo(Boolean.toString(true));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_VALUE))
+        .isEqualTo(executionPayloadValue.toDecimalString());
+    final String body = response.body().string();
+    assertThat(body).isEqualTo(getExpectedBlockAsJson(specMilestone, true, true));
+  }
+
+  @TestTemplate
+  void shouldGetBlindedBlockContentAsJsonPostDenebAsSsz() throws IOException {
+    assumeThat(specMilestone).isEqualTo(DENEB);
+    final BlindedBlockContents blindedBlockContents =
+        dataStructureUtil.randomBlindedBlockContents(ONE);
+    final BLSSignature signature = blindedBlockContents.getBlock().getBody().getRandaoReveal();
+    when(validatorApiChannel.createUnsignedBlock(eq(UInt64.ONE), eq(signature), any()))
+        .thenReturn(SafeFuture.completedFuture(Optional.of(blindedBlockContents)));
+    when(executionLayerBlockProductionManager.getCachedPayloadResult(UInt64.ONE))
+        .thenReturn(
+            Optional.of(
+                new ExecutionPayloadResult(
+                    mock(ExecutionPayloadContext.class),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.of(SafeFuture.completedFuture(executionPayloadValue)))));
+    Response response = get(signature, ContentTypes.OCTET_STREAM);
+    assertThat(response.code()).isEqualTo(SC_OK);
+    assertThat(response.header(HEADER_CONSENSUS_VERSION))
+        .isEqualTo(specMilestone.name().toLowerCase(Locale.ROOT));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_BLINDED)).isEqualTo(Boolean.toString(true));
+    assertThat(response.header(HEADER_EXECUTION_PAYLOAD_VALUE))
+        .isEqualTo(executionPayloadValue.toDecimalString());
+    final BlindedBlockContents result =
+        (BlindedBlockContents)
+            spec.getGenesisSchemaDefinitions()
+                .getBlindedBlockContainerSchema()
+                .sszDeserialize(Bytes.of(response.body().bytes()));
+    assertThat(result).isEqualTo(blindedBlockContents);
+  }
+
+  private Response get(final BLSSignature signature, final String contentType) throws IOException {
+    return getResponse(
+        GetNewBlockV3.ROUTE.replace("{slot}", "1"),
+        Map.of("randao_reveal", signature.toString()),
+        contentType);
+  }
+
+  private String getExpectedBlockAsJson(
+      final SpecMilestone specMilestone, final boolean blinded, final boolean blockContents)
+      throws IOException {
+    final String fileName =
+        String.format(
+            "new%s%s%s.json",
+            blinded ? "Blinded" : "",
+            blockContents ? "BlockContents" : "Block",
+            specMilestone.name());
+    return Resources.toString(
```

Comment:
I'd better parse json. Every time we change something in dataStructureUtil order could be changed and all tests with hardcoded random expectation starts to failing. Block structures are big and this means more chances to affect it. Plus we have the actual json data by the hand. I'd propose to refactor this to json parsing. @tbenr what's your feeling?

Reply:
As discussed, we didn't add the annotated classes for the json parser and hence can't parse the generated block objects. We could refactor the tests in a separate PR if we agree to do so